Kafka: failure to propagate errors when poll fails

@devinrsmith :
> I'm able to configure kafka in a bad way and not get a table error notification in the web UI (making me think something is hooked up wrong).
> might be that we aren't handling exceptions at the wire deser layer:
```
16:08:21.346 ERROR tertopic-test-11:ALL | i.d.kafka.KafkaTools      | KafkaIngester(topic-test-11, ALL): Exception while polling for Kafka messages:org.apache.kafka.common.errors.RecordDeserializationException: Error deserializing key/value for partition topic-test-11-0 at offset 0. If needed, please seek past the record to continue consumption.
	at org.apache.kafka.clients.consumer.internals.Fetcher.parseRecord(Fetcher.java:1435)
	at org.apache.kafka.clients.consumer.internals.Fetcher.access$3400(Fetcher.java:133)
	at org.apache.kafka.clients.consumer.internals.Fetcher$CompletedFetch.fetchRecords(Fetcher.java:1658)
	at org.apache.kafka.clients.consumer.internals.Fetcher$CompletedFetch.access$1900(Fetcher.java:1494)
	at org.apache.kafka.clients.consumer.internals.Fetcher.fetchRecords(Fetcher.java:716)
	at org.apache.kafka.clients.consumer.internals.Fetcher.collectFetch(Fetcher.java:682)
	at org.apache.kafka.clients.consumer.KafkaConsumer.pollForFetches(KafkaConsumer.java:1318)
	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1247)
	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1220)
	at io.deephaven.kafka.ingest.KafkaIngester.pollOnce(KafkaIngester.java:340)
	at io.deephaven.kafka.ingest.KafkaIngester.consumerLoop(KafkaIngester.java:297)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.apache.kafka.common.errors.SerializationException: Size of data received by LongDeserializer is not 8
	at org.apache.kafka.common.serialization.LongDeserializer.deserialize(LongDeserializer.java:26)
	at org.apache.kafka.common.serialization.LongDeserializer.deserialize(LongDeserializer.java:21)
	at org.apache.kafka.common.serialization.Deserializer.deserialize(Deserializer.java:60)
	at org.apache.kafka.clients.consumer.internals.Fetcher.parseRecord(Fetcher.java:1426)
	... 11 more
, aborting.
16:08:21.346 ERROR tertopic-test-11:ALL | i.d.kafka.KafkaTools      | KafkaIngester(topic-test-11, ALL): Stopping due to errors (0 messages with error out of 0 messages processed)
```
> this is if I give it an incompatible / wrong org.apache.kafka.common.serialization.Deserializer
> yep; and table.isFailed() is false

The root cause is missing failure propagation in this case. Should be a small change.

--- a/extensions/kafka/src/main/java/io/deephaven/kafka/ingest/KafkaIngester.java
+++ b/extensions/kafka/src/main/java/io/deephaven/kafka/ingest/KafkaIngester.java
@@ -185,7 +185,7 @@ public class KafkaIngester {
 
     /**
      * Creates a Kafka ingester for the given topic.
-     * 
+     *
      * @param log A log for output
      * @param props The properties used to create the {@link KafkaConsumer}
      * @param topic The topic to replicate
@@ -331,6 +331,13 @@ public class KafkaIngester {
         } catch (Exception ex) {
             log.error().append(logPrefix).append("Exception while polling for Kafka messages:").append(ex)
                     .append(", aborting.").endl();
+            final KafkaRecordConsumer[] allConsumers;
+            synchronized (streamConsumers) {
+                allConsumers = streamConsumers.valueCollection().toArray(KafkaRecordConsumer[]::new);
+            }
+            for (final KafkaRecordConsumer streamConsumer : allConsumers) {
+                streamConsumer.acceptFailure(ex);
+            }
             return false;
         }
 

